### PR TITLE
Oauth save option

### PR DESCRIPTION
**Pull request by aashish-jain**
Added functionality to save tokens on the front-end.

### CODE REVIEW

**Comment by aashish-jain** [comment]
[OAuth cancel scenario](https://trello.com/c/DzVnJCnq/7-oauth-cancel-scenario)

**Comment by aashish-jain** [comment]
[OAuth storage preference](https://trello.com/c/UnnQ1cjI/10-oauth-storage-preference)

**Comment by aashish-jain** [comment]
[Account preference pop up](https://trello.com/c/bJZQ8ukd/6-account-preference-pop-up)

**Review by LinusCastelino — COMMENTED** [review]
@aashish-jain 
If you have time to spare, please create javadoc comments for every backend method you created or edited.
I can help if required.

**Comment by aashish-jain** [comment]
@LinusCastelino added the javadoc comments for the controllers

File: src/main/java/org/onedatashare/server/service/oauth/OauthService.java [hunk 1]
```diff
@@ -19,79 +19,86 @@
...
+
+    public RedirectView redirectToDropboxAuth(Boolean value) {
+        String url = start();
+        return new RedirectView(url);
+    }
+
+    @PostConstruct
+    public void postConstructInit() {
+        secrets = new DbxAppInfo(this.key, this.secret);
+        config = DbxRequestConfig.newBuilder("OneDataShare-DIDCLab").build();
```
Comment (LinusCastelino) [inline src/main/java/org/onedatashare/server/service/oauth/OauthService.java:89]: Move the application name to ODS Constants

File: src/main/java/org/onedatashare/server/model/useraction/UserAction.java [hunk 2]
```diff
@@ -3,41 +3,121 @@
 import com.fasterxml.jackson.annotation.JsonInclude;
```
Comment (LinusCastelino) [inline src/main/java/org/onedatashare/server/model/useraction/UserAction.java:3]: Please add javadocs here

**Review by LinusCastelino — CHANGES_REQUESTED** [review]

**Review by LinusCastelino — COMMENTED** [review]
Approving changes since the PR is huge.
Issues raised out of this PR will be treated as bugs.

**Pull request merged**